src/main/java/tfifteenfour/clipboard/model/course/Course.java

```diff
@@ -0,0 +1,96 @@
+package tfifteenfour.clipboard.model.course;
+
+import static java.util.Objects.requireNonNull;
+import static tfifteenfour.clipboard.commons.util.AppUtil.checkArgument;
+
+import javafx.collections.ObservableList;
+
+/**
+ * Represents a Course in the CLIpboard.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidCourseCode(String)} (String)}
+ */
+public class Course {
+
+    public static final String MESSAGE_CONSTRAINTS = "Course codes should be alphanumeric";
+    public static final String VALIDATION_REGEX = "\\p{Alnum}+";
+
+    public final String courseCode;
+	private UniqueGroupsList groups;
+
+    {
+        groups = new UniqueGroupsList();
+    }
```

Comment:
is there a reason why `groups = new UniqueGroupsList();` needs to be written separately? 🤔

Reply:
I actually have no idea why. But it's the way UniqueStudentList from AB3 was instantiated also, and since UniqueGroupList is just a mirror of it, I just kept it in case theres a special reason why they did it for AB3. 

There are some comments that AB3 dev talked about it in UniqueStudentList

Reply:
Something to take note of if yall decide to implement similar lists for Attendance/Assignments 

I actually found it a little dangerous as I encountered a nasty bug where my data was overwritten by this upon first loading from storage, so ended up with empty data. Need to remember to call the resetData() when the object is instantiated.

Something to keep in mind, for @IsabelChong also.

Reply:
😮 I see, thanks for the explanation!

For other's reference, here's the comment:

> The 'unusual' code block below is a non-static initialization block, sometimes used to avoid duplication
> between constructors. See https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html
> 
> Note that non-static init blocks are not recommended to use. There are other ways to avoid duplication
> among constructors.